ITensorGaussianMPS/src/gmps.jl

```diff
@@ -292,16 +660,47 @@ function correlation_matrix_to_mps(
   Λ::AbstractMatrix;
   eigval_cutoff::Float64=1e-8,
   maxblocksize::Int=size(Λ, 1),
+  minblocksize::Int=1,
   kwargs...,
 )
-  @assert size(Λ, 1) == size(Λ, 2)
+  return correlation_matrix_to_mps(
+    s,
+    ConservingNf(Λ0);
+    eigval_cutoff=eigval_cutoff,
+    maxblocksize=maxblocksize,
+    minblocksize=minblocksize,
+    kwargs...,
+  )
+end
+
+function correlation_matrix_to_mps(
+  s::Vector{<:Index},
+  Λ0::T;
+  eigval_cutoff::Float64=1e-8,
+  maxblocksize::Int=size(Λ0.data, 1),
+  minblocksize::Int=1,
+  kwargs...,
+) where {T<:Union{Pairing,ConservingNf}}
+  if eltype(Λ0.data) <: AbstractFloat
+    MPS_Elt = Float64
+  else
+    MPS_Elt = ComplexF64
+  end
+  Λ = check_pairing_correlations(Λ0)
+  calctype = typeof(Λ)
+  @assert size(Λ.data, 1) == size(Λ.data, 2)
   ns, C = correlation_matrix_to_gmps(
-    Λ; eigval_cutoff=eigval_cutoff, maxblocksize=maxblocksize
+    Λ; eigval_cutoff=eigval_cutoff, minblocksize=minblocksize, maxblocksize=maxblocksize
   )
   if all(hastags("Fermion"), s)
-    U = [ITensor(s, g) for g in reverse(C.rotations)]
-    ψ = MPS(s, n -> round(Int, ns[n]) + 1, U; kwargs...)
+    U = to_manybody_gates(s, calctype(C))
+    ψ = MPS(MPS_Elt, s, n -> round(Int, ns[calctype <: Pairing ? 2 * n : n]) + 1)
```

Comment:
Again, I would use a generic `site_stride` function here instead of an if-statement.